cli/tools/bundle/html.rs

```diff
@@ -0,0 +1,358 @@
+        Child::Comment(c) => self.visit_comment(c),
+        Child::DocumentType(d) => self.visit_document_type(d),
+      }
+    }
+  }
+}
+
+pub mod walk {
+  use swc_html::ast::Element;
+
+  use super::VisitHtml;
+
+  pub fn element(elem: &mut Element, visitor: &mut impl VisitHtml) {
+    for e in &mut elem.children {
+      match e {
+        swc_html::ast::Child::DocumentType(document_type) => {
+          visitor.visit_document_type(document_type)
+        }
+        swc_html::ast::Child::Element(element) => {
+          visitor.visit_element(element)
+        }
+        swc_html::ast::Child::Text(text) => visitor.visit_text(text),
+        swc_html::ast::Child::Comment(comment) => {
+          visitor.visit_comment(comment)
+        }
+      }
+    }
+  }
+}
+
+struct Visitor {
+  scripts: Vec<Script>,
+}
+
+fn get_attr<'a>(e: &'a Element, name: &str) -> Option<&'a str> {
+  e.attributes
+    .iter()
+    .find(|a| a.name == name)
+    .and_then(|a| a.value.as_ref().map(|v| v.as_str()))
+}
+
+impl VisitHtml for Visitor {
+  fn visit_element(&mut self, e: &mut Element) {
+    if e.tag_name == "script" {
+      let src = get_attr(e, "src");
+      let typ = get_attr(e, "type");
+      if let (Some(src), Some(typ @ "module")) = (src, typ) {
+        self.scripts.push(Script {
+          src: Some(src.to_string()),
+          is_async: get_attr(e, "async").is_some(),
+          is_module: typ == "module",
+          is_ignored: get_attr(e, "deno-ignore").is_some()
+            || get_attr(e, "vite-ignore").is_some(),
+          resolved_path: None,
+        });
+      }
+    }
+    walk::element(e, self);
+  }
+}
+
+fn collect_scripts(doc: &mut Document) -> Vec<Script> {
+  let mut visitor = Visitor { scripts: vec![] };
+  visitor.visit_document(doc);
+  visitor.scripts.retain(|s| !s.is_ignored);
+  visitor.scripts
+}
+
+#[derive(Debug, Clone)]
+pub struct HtmlEntrypoint {
+  pub path: PathBuf,
+  pub scripts: Vec<Script>,
+  pub temp_module: String,
+  pub doc: Document,
+}
+
+pub fn doit(path: &Path) -> anyhow::Result<HtmlEntrypoint> {
+  let file: Rc<deno_ast::swc::common::FileName> =
+    Rc::new(PathBuf::from(path).into());
+  let file = SourceFile::new(
+    file.clone(),
+    false,
+    file.clone(),
+    bytes_str::BytesStr::from_utf8_vec(std::fs::read(path)?.into())?,
+    BytePos(1),
+  );
+  let mut errors = Vec::new();
+  let mut doc = swc_html::parser::parse_file_as_document(
+    &file,
+    swc_html::parser::parser::ParserConfig {
+      ..Default::default()
+    },
+    &mut errors,
+  )
+  .unwrap();
+
+  let mut scripts = collect_scripts(&mut doc);
+
+  let mut temp_module = String::new();
+  for script in &mut scripts {
+    if let Some(src) = &mut script.src {
+      let src = src.trim_start_matches('/');
+      let path = path.parent().unwrap().join(src);
+
+      temp_module
+        .push_str(&format!("import \"{}\";\n", path.to_string_lossy()));
+      script.resolved_path = Some(path);
+    }
+  }
```

Comment:
You probably want to use `capacity_builder::StringBuilder` here, but a comment regarding what is this temp module might be very helpful